### DISCUSSION TITLE
Naming a run

### DISCUSSION THREAD

**Discussion by ReubenTheDS** [PrefectHQ/prefect#3881] [Q&A]
Hi team, how do I name a run programmatically? I am able to name a flow & have seen [the idiom page](https://docs.prefect.io/core/idioms/idioms.html) for naming tasks.

Background:
I have a scheduled flow that runs daily midnight, that I can see & interact with, on my localhost Prefect server using a Server project ```"My_Proj"```
```
schedule = CronSchedule("0 0 * * *", start_date=pendulum.now())
with Flow("My_Flow", schedule=schedule) as flow:
    res1 = cool_task1()
    res2 = cool_task2( res1 )
    # more tasks here

flow.register(project_name="My_Proj")
```

I see ( very cool :-)  ) named runs like "tentacled-tanuki", "indigo-badger", "lilac-earwig", "asparagus-badger", etc.
Using the UI, I can trigger a Run and name it.  Programmatically, I am now aware that I can use prefect.context to access the run-date. My question is:
How do set up my code in Python to **name** a run based on that day's date?

**Comment by cicdw** [comment]
Hi @ReubenTheDS - you can use the `set_flow_run_name` method of the Prefect `Client` for this; where you call this is totally up to you, but if it were me I'd use a state handler like so:

```python
import prefect


def set_run_name(flow, old_state, new_state):
    if new_state.is_running():
        client = prefect.Client()
        name = f"{flow.name}-{prefect.context.date:%A}" # use flow-name-day-of-week as the flow run name, for example
        client.set_flow_run_name(prefect.context.flow_run_id, name)


with Flow(..., state_handlers=[set_run_name]) as flow:
    ...
```

But you could place that client call anywhere you'd like (within the first task, etc.)

**Reply by ReubenTheDS** [reply]
Thank you so much, Chris.
